cloud/config/config.go

```diff
@@ -83,6 +88,10 @@ func getConfigFromFile(path string) (*AuthConfig, error) {
 		return nil, errors.Wrap(err, "unmarshalling auth config")
 	}
 
+	if err := deferCloseFunc(); err != nil {
```

Comment:
Going by the wiki, we should just do defer deferCloseFunc right? Otherwise, if we get an error in say line 84, we will never close the file, if that is the writable case which it is not.